### ISSUE TITLE
App in production started crashing suddenly.  Updated schema has 2 entity types, but incoming schema has 2.

### ISSUE THREAD

**Issue by binishmatheww** [objectbox/objectbox-java#1156]
Hi all,

A new version of our app was rolled out 2 months ago and it was working fine till yesterday.
All of a sudden, we started seeing a spike with this issue on Crashlytics. 
Attaching the stacktrace below

```
Caused by java.lang.IllegalStateException: Updated schema has 2 entity types, but incoming schema has 2
       at io.objectbox.BoxStore.nativeCreateWithFlatOptions(BoxStore.java)
       at io.objectbox.BoxStore.<init>(BoxStore.java:268)
       at io.objectbox.BoxStoreBuilder.build(BoxStoreBuilder.java:549)
       at com.moveinsync.driver.byod.datamodels.DatabaseManager.<init>(DatabaseManager.java:88)
       at com.moveinsync.driver.byod.utils.HealthMonitor.initialize(HealthMonitor.java:71)
       at com.moveinsync.driver.byod.modules.MyApplication.initAfterSiteSelection(MyApplication.kt:204)
       at com.moveinsync.driver.byod.modules.MyApplication.onCreate(MyApplication.kt:163)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1266)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7598)
       at android.app.ActivityThread.-$$Nest$mhandleBindApplication()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2397)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8741)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

Any help is much appreciated.
Thanks.

**Comment by binishmatheww** [comment]
@greenrobot please look into this issue. Thanks.

**Comment by greenrobot-team** [comment]
Thanks for reporting! This error message occurs when the database has not correctly updated its schema with the new one given to it, e.g. when modifying/adding or removing entities. At this point I'm not sure when this could happen, but maybe this gives you a hint where to look.

We could really benefit from you supplying as many details as possible as indicated by [the bug report template](https://github.com/objectbox/objectbox-java/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=) to let us isolate this issue.

**Comment by greenrobot-team** [comment]
@binishmatheww Do you have any additional information for us?

Note that we have released 3.7.1 which fixes the error message to show the actual count of expected entities of the incoming schema.
